builder/amazon/common/access_config.go

```diff
@@ -34,14 +38,78 @@ func (c *AccessConfig) Session() (*session.Session, error) {
 		return c.session, nil
 	}
 
-	config := aws.NewConfig().WithMaxRetries(11).WithCredentialsChainVerboseErrors(true)
+	// build a chain provider, lazy-evaluated by aws-sdk
+	providers := []credentials.Provider{
+		&credentials.StaticProvider{Value: credentials.Value{
+			AccessKeyID:     c.AccessKey,
+			SecretAccessKey: c.SecretKey,
+			SessionToken:    c.Token,
+		}},
+		&credentials.EnvProvider{},
+		&credentials.SharedCredentialsProvider{
+			Filename: "",
+			Profile:  c.ProfileName,
+		},
+	}
+
+	// Build isolated HTTP client to avoid issues with globally-shared settings
+	client := cleanhttp.DefaultClient()
 
-	if c.ProfileName != "" {
-		if err := os.Setenv("AWS_PROFILE", c.ProfileName); err != nil {
-			return nil, fmt.Errorf("Set env error: %s", err)
+	// Keep the default timeout (100ms) low as we don't want to wait in non-EC2 environments
+	client.Timeout = 100 * time.Millisecond
+
+	const userTimeoutEnvVar = "AWS_METADATA_TIMEOUT"
+	userTimeout := os.Getenv(userTimeoutEnvVar)
```

Comment:
I think we do because

a) doesn't describe an artifact, but instead is implementation related
b) there's prior art (see below)
c) it provides compatibility with terraform, and I think a more consistent experience is a good thing (even tho we have a ways to go in that regard)

```
builder/amazon/common/state.go: override := os.Getenv("AWS_TIMEOUT_SECONDS")
builder/amazon/common/state.go: override := os.Getenv("AWS_POLL_DELAY_SECONDS")
```

Reply:
Fair! Those are all great reasons.